src/components/customers/AddCustomerDrawer.tsx

```diff
@@ -49,6 +50,14 @@ export const AddCustomerDrawer = forwardRef<DrawerRef, AddCustomerDrawerProps>(
     const { isEdition, onSave } = useCreateEditCustomer({
       customer,
     })
+    const [isDisabled, setIsDisabled] = useState<boolean>(false)
+    const handleChange = (checked: boolean) => {
```

Comment:
Probably you can move this method directly into the checkbox onChange function.

Also `isDisabled` could check on `providerCustomer.syncWithProvider` value directly instead of being a `useState`

Reply:
Fixed first only first part (as discussed offline)